clients/src/main/java/org/apache/kafka/clients/producer/RecordMetadata.java

```diff
@@ -54,15 +57,7 @@ private RecordMetadata(TopicPartition topicPartition, long offset, long timestam
 
     @Deprecated
     public RecordMetadata(TopicPartition topicPartition, long baseOffset, long relativeOffset) {
```

Comment:
Yeah, I was thinking it was fine to remove it since users should not create an instance of this apart from tests and it's been deprecated for a few release cycles. Or if we think the method is useful, then it would make sense undeprecate it.